src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -918,6 +934,46 @@ map_injective f.injective u v
 
 end path
 
+/-! ## Deleting edges -/
+
+namespace walk
+variables {G}
+
+/-- Given a walk that avoids a set of edges, produce a walk in the graph
+with those edges deleted. -/
+def to_delete_edges (s : set (sym2 V)) :
+  Π {v w : V} (p : G.walk v w) (hp : ∀ e, e ∈ p.edges → ¬ e ∈ s), (G.delete_edges s).walk v w
+| _ _ nil _ := nil
+| _ _ (cons' u v w h p) hp := cons' u v w (by simpa [h, -quotient.eq] using hp ⟦(u, v)⟧)
```

Comment:
This pattern seems to happen reasonably often when working with sym2 - is it possible for us to add a sym2.eq lemma which would both take precedence over this one, and produce helpful results? I suppose the other solution is to remove `quotient.eq` as a simp lemma but that seems undesirable to me. Another option is to have the right hand side of `quotient.eq` have its own simp lemma for sym2?

Reply:
I've been imagining that, once the `quotient` notation becomes a notation typeclass, we'd give `sym2` its own quot api to solve this once and for all, though there are probably other options, too.

There's already `sym2.eq_iff` that's better than `quotient.eq`. I think if we made a `simp` lemma that's that but with the setoid relation on the LHS we might not need `-quotient.eq` in many places.

Reply:
Right, I think your last suggestion sounds super helpful

Reply:
Ok, #14113 adds that lemma (not sure if it cleanly compiles yet).

Reply:
`-quotient.eq`'s have now been eliminated